src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationProvider.cs

```diff
@@ -221,7 +229,7 @@ public async Task RefreshAsync(CancellationToken cancellationToken)
                     StringBuilder logInfoBuilder = new StringBuilder();
                     StringBuilder logDebugBuilder = new StringBuilder();
 
-                    await ExecuteWithFailOverPolicyAsync(availableClients, async (client) =>
+                    await ExecuteWithFailOverPolicyAsync(availableClients.GetAsyncEnumerator(), async (client) =>
```

Comment:
It has to be the same enumerator that was checked or else there's no value to the change.

Reply:
I haven't completed the work

Reply:
You are right, I'm really updating it